app/scripts/background.js

```diff
@@ -94,16 +95,47 @@ const WORKER_KEEP_ALIVE_MESSAGE = 'WORKER_KEEP_ALIVE_MESSAGE';
 /**
  * In case of MV3 we attach a "onConnect" event listener as soon as the application is initialised.
  * Reason is that in case of MV3 a delay in doing this was resulting in missing first connect event after service worker is re-activated.
+ *
+ * @param remotePort
  */
-
 const initApp = async (remotePort) => {
   browser.runtime.onConnect.removeListener(initApp);
   await initialize(remotePort);
   log.info('MetaMask initialization complete.');
 };
 
+/**
+ * Sends a message to the dapp(s) content script to signal it can connect to MetaMask background as
+ * the backend is not active. It is required to re-connect dapps after service worker re-activates.
+ * For non-dapp pages, the message will be sent and ignored.
+ */
+const sendReadyMessageToTabs = async () => {
+  const tabs = await browser.tabs
+    .query({
+      // Query tabs with a url. New tabs and DevTool tabs have no url value. e.g. chrome://extensions, chrome://inspect
+      url: '<all_urls>',
```

Comment:
The docs seem to indicate that this is ignored if we don't have the `tabs` permission: https://developer.chrome.com/docs/extensions/reference/tabs/#method-query

Reply:
It seems so from the docs, but it appears to work. 

I updated the code with a comment as to how and why it works:
```
      /**
       * Only query tabs that our extension can run in. To do this, we query for all URLs that our
       * extension can inject scripts in, which is by using the "<all_urls>" value and __without__
       * the "tabs" manifest permission. If we included the "tabs" permission, this would also fetch
       * URLs that we'd not be able to inject in, e.g. chrome://pages, chrome://extension, which
       * is not what we'd want.
       *
       * You might be wondering, how does the "url" param work without the "tabs" permission?
       *
       * @see {@link https://bugs.chromium.org/p/chromium/issues/detail?id=661311#c1}
       *  "If the extension has access to inject scripts into Tab, then we can return the url
       *   of Tab (because the extension could just inject a script to message the location.href)."
       */
```

Thanks for pointing this out @Gudahtt. I didn't fully realize why the `chrome://` URLs weren't being returned here with `url: '<all_urls>',` until I dug into this more. It was difficult to find documentation that elaborated on this, but the [comments written by a chromium dev (above @link)](https://bugs.chromium.org/p/chromium/issues/detail?id=661311#c1) helped.

Reply:
note: I attempted to add a comment into the manifest.json, but unfortunately, this causes a warning found through chrome://extensions > Errors

<img width="418" alt="Screen Shot 2022-11-08 at 00 18 54" src="https://user-images.githubusercontent.com/20778143/200422988-35afe371-81c8-4b04-a9ac-ebbd1ab0eee5.png">

Reply:
Great find @digiwand ! And I appreciate the detailed comment, that's very helpful